Interpose.Core/Generators/CachedTypeGenerator.cs

```diff
@@ -35,9 +34,9 @@ private void Register(IInterceptor interceptor, Type type, Type baseType, Type[]
             this.proxyTypes[key] = type;
         }
 
-        private string GenerateKey(Type baseType, Type[] additionalInterfaceTypes, Type handlerType)
+        private string GenerateKey(Type baseType, IEnumerable<Type> additionalInterfaceTypes, Type handlerType)
```

Comment:
This is kind of irrelevant as this is a private method that is only ever called in a single way.

Reply:
Ah, sorry, went over the top with ReSharper's code suggestions

Reply:
Thanks! :-)